hypertrace-trace-enricher/trace-reader/src/main/java/org/hypertrace/trace/reader/attributes/TraceAttributeReader.java

```diff
@@ -1,18 +1,13 @@
 package org.hypertrace.trace.reader.attributes;
 
 import io.reactivex.rxjava3.core.Single;
-import org.hypertrace.core.attribute.service.cachingclient.CachingAttributeClient;
+import org.apache.avro.generic.GenericRecord;
 import org.hypertrace.core.attribute.service.v1.LiteralValue;
-import org.hypertrace.core.datamodel.Event;
-import org.hypertrace.core.datamodel.StructuredTrace;
 
-public interface TraceAttributeReader {
-  Single<LiteralValue> getSpanValue(
-      StructuredTrace trace, Event span, String attributeScope, String attributeKey);
+public interface TraceAttributeReader<T extends GenericRecord, S extends GenericRecord> {
+  Single<LiteralValue> getSpanValue(T trace, S span, String attributeScope, String attributeKey);
 
-  Single<LiteralValue> getTraceValue(StructuredTrace trace, String attributeKey);
+  Single<LiteralValue> getTraceValue(T trace, String attributeKey);
 
-  static TraceAttributeReader build(CachingAttributeClient attributeClient) {
-    return new DefaultTraceAttributeReader(attributeClient);
-  }
+  String getCustomerId(S span);
```

Comment:
getTenantId (even if it's still called customerId due to tech debt, no need to spread it)